packages/terra-brand-footer/src/BrandFooter.jsx

```diff
@@ -40,35 +94,81 @@ const propTypes = {
 
 const defaultProps = {
   links: [],
+  sections: {
+    displayVertically: false,
+    linkGroups: [],
+  },
   contentLeft: null,
   contentRight: null,
   contentBottom: null,
 };
 
-const BrandFooter = ({ links, contentLeft, contentRight, contentBottom, ...customProps }) => {
+const BrandFooter = ({ links, sections, contentLeft, contentRight, contentBottom, ...customProps }) => {
   const BrandFooterClassNames = cx([
     'brand-footer',
     customProps.className,
   ]);
 
-  const linkslength = links.length;
+  const processedSections = {};
+  if (links.length > 0 && sections.linkGroups.length === 0) {
+    processedSections.displayVertically = false;
+    processedSections.linkGroups = [{
+      headerText: '',
+      links,
+      id: 0,
+    }];
+  } else {
+    processedSections.displayVertically = sections.displayVertically;
+    processedSections.linkGroups = sections.linkGroups;
+
+    // Assign ids to use as keys
+    for (let i = 0; i < processedSections.linkGroups.length; i += 1) {
+      processedSections.linkGroups[i].id = i;
+    }
+  }
+
+  // The old links prop can't have section headers. Needed for vertical layout to insert padding to keep columns without headers aligned
+  const containsASectionHeader = processedSections.linkGroups.some(linkGroup => linkGroup.headerText);
+
   return (
     <footer role="contentinfo" {...customProps} className={BrandFooterClassNames}>
-      {linkslength > 0 ? (
-        <nav className={cx('nav')}>
-          <ul className={cx('menu')}>
-            {links.map(link => (
-              <li className={cx('list-item')} key={link.text + link.href}>
-                {link.target !== undefined ? (
-                  <a className={cx('link')} href={link.href} target={link.target} >{link.text}</a>
-                ) : (
-                  <a className={cx('link')} href={link.href} >{link.text}</a>
+      {processedSections.linkGroups.length > 0 && (
+        <nav className={cx(['nav', !processedSections.displayVertically ? 'nav-horizontal' : 'nav-vertical'])}>
+            {processedSections.linkGroups.map(linkGroup => (
+              <ul className={cx('menu')} key={linkGroup.id}>
+                { // When displaying vertically if one column has a header all columns are aligned as if they have a header
+                  ((containsASectionHeader && processedSections.displayVertically) || linkGroup.headerText) && (
+                    <li className={cx('list-header')} key={linkGroup.headerText}>
+                      { // Insert a zero width space to act as a placeholder section header that doesn't display but takes vertical space
```

Comment:
In regards to this comment:
>// Insert a zero width space to act as a placeholder section header that doesn't display but takes vertical space
why do we want to do this?

Reply:
In the design specification link sections without a section header were shown to keep their first element aligned vertically with other link sections that do have a section header. The zero width space takes up as much vertical room as regular text would.